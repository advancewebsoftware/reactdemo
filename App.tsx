import 'react-native-gesture-handler';
import {store} from './src';
import {Colors} from './src/theme';
import {Provider} from 'react-redux';
import {StatusBar} from 'react-native';
import React, {useEffect} from 'react';
import {StackNavigation} from './src/navigators';
import SplashScreen from 'react-native-splash-screen';

const App: React.FC = () => {
  useEffect(() => {
    SplashScreen?.hide();
  }, []);

  return (
    <Provider store={store}>
      <StatusBar
        barStyle={'light-content'}
        backgroundColor={Colors.headerBackground}
      />
      <StackNavigation />
    </Provider>
  );
};

export default App;
