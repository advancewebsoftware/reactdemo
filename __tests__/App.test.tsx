import 'react-native';
import App from '../App';
import React from 'react';
import {render} from '@testing-library/react-native';

jest.mock('@react-navigation/stack', () => ({
  createStackNavigator: () => ({
    Navigator: 'mocked-Navigator',
    Screen: 'mocked-Screen',
  }),
  TransitionPresets: jest.fn(),
}));

jest.mock('react-native-gesture-handler', () => {});
jest.mock('react-native-webview', () => {});
jest.mock('react-native-splash-screen', () => {});

jest.mock('react-redux', () => ({
  useDispatch: jest.fn(),
  useSelector: jest.fn().mockReturnValue({}),
  Provider: ({children}: any) => children,
}));

jest.mock('@reduxjs/toolkit', () => ({
  createAsyncThunk: jest.fn(),
  createSlice: jest.fn().mockImplementation(options => ({
    ...options,
    reducer: jest.fn(),
    actions: {},
  })),
  combineReducers: jest.fn(),
  configureStore: jest.fn(),
}));

jest.mock('redux', () => ({
  combineReducers: jest.fn(),
  applyMiddleware: jest.fn(),
  createStore: jest.fn(),
}));

const component = <App />;

describe('App', () => {
  it('renders correctly', () => {
    const container = render(component);
    expect(container.toJSON()).toMatchSnapshot();
  });
});
