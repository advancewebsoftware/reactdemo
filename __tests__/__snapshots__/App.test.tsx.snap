// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<mocked-Navigator
  screenOptions={
    {
      "headerShown": false,
    }
  }
>
  <mocked-Screen
    component={[Function]}
    name="Home"
  />
  <mocked-Screen
    component={[Function]}
    name="WebView"
    options={
      {
        "gestureEnabled": true,
        "headerShown": false,
      }
    }
  />
</mocked-Navigator>
`;
